internal/codespaces/api/api.go

```diff
@@ -268,19 +268,37 @@ func (c *Codespace) ExportData(fields []string) map[string]interface{} {
 	return data
 }
 
+type ListCodespacesOptions struct {
+	OrgName    string
+	UserName   string
+	Repository *Repository
+	Limit      int
+}
+
 // ListCodespaces returns a list of codespaces for the user. Pass a negative limit to request all pages from
 // the API until all codespaces have been fetched.
-func (a *API) ListCodespaces(ctx context.Context, limit int, orgName string, userName string) (codespaces []*Codespace, err error) {
-	perPage := 100
+func (a *API) ListCodespaces(ctx context.Context, opts ListCodespacesOptions) (codespaces []*Codespace, err error) {
+
+	var (
+		perPage = 100
+		limit   = opts.Limit
+	)
+
 	if limit > 0 && limit < 100 {
 		perPage = limit
 	}
 
 	var listURL string
 	var spanName string
 
-	if orgName != "" {
-		if userName != "" {
+	if opts.Repository != nil {
```

Comment:
Should there be some validation that you shouldn't set both `Repository` and `OrgName` at the same time? Otherwise if you set both, `OrgName` (and `UserName` for that matter) get silently ignored.

Reply:
Oh, I see in the command that this is enforced in the args. Do you think it makes sense to still add validation here so that other commands don't make this mistake later?

Reply:
yeah I chose to do it where the args are parsed from the CLI. With the current logic even if they are both set (or some other permutation) it will still only hit the `/repo` path since its first in the `if/else` block